rdtools/filtering.py

```diff
@@ -743,3 +743,35 @@ def xgboost_clip_filter(power_ac,
                       & (power_ac_df['scaled_value'] >= .1))
     final_clip = final_clip.reindex(index=power_ac.index, fill_value=False)
     return ~(final_clip.astype(bool))
+
+
+def hampel_filter(vals, k=14, t0=3):
+    '''
+    Hampel outlier filter primarily applied on daily normalized data but broadly
+    applicable.
+
+    Parameters
+    ----------
+    vals : pandas.Series
+        daily normalized time series
+    k : int, default 14
+        size of window including the sample; 14 is equal to 7 on either side of
+        value
+    t0 : int, default 3
+        Threshold value, defaults to 3 sigma Pearson's rule.
+
+    Returns
+    -------
+    pandas.Series
+        Boolean Series of whether the given measurement is within 3 sigma of the
+        median.  False points indicate outliers to be removed.
+    '''
+    # Make copy so original not edited
+    vals_copy = vals.copy()
+    # Hampel Filter
+    L = 1.4826
+    rolling_median = vals_copy.rolling(k, center=True, min_periods=1).median()
```

Comment:
On our last RdTools call I said I thought `rolling(k, center=True)` didn't work if `k` was a pandas offset instead of an integer, but it works for me with pandas 1.4.2 so I retract that point.  (No change needed here, just wanted to correct my misstatement)

Reply:
Note that I tried setting k = '14d' as a default value in filtering.hampel_filter, but it resulted in tests failing.  This may have more to do with how the tests were written vs anything fundamentally incompatible.

Reply:
I suggest we get to the bottom of why the tests failed before giving up on the time-based default window.

Reply:
OK, here's some analysis of the different options.  Note that there is a very different result between 14 and '14d'.  In general, using '14d' leads to the incorrect dates being selected as outliers, often the date selected by '14d' is the day prior to when 14 would indicate a NaN, but not always.  This is the output of the test case test_sensor_analysis_hampel_filter() plotted in excel.  The points indicate where each outlier option indicates a NaN value.
![image](https://user-images.githubusercontent.com/23244088/216731544-7b8bf55f-893c-4edf-be21-8137945337ad.png)

Reply:
Here's the full difference spreadsheet
[hampel_comparison.xlsx](https://github.com/NREL/rdtools/files/10586562/hampel_comparison.xlsx)

Reply:
OK, when I run this on the example notebook, the difference between the two approaches is very similar.  I guess when you have 'actual' outlier data, they are both doing well.  Maybe '14d' is slightly better at removing outlier points.

k='14d' plot.  N=1991 remaining points
![image](https://user-images.githubusercontent.com/23244088/217383621-073f1d57-15d0-4092-a956-f2d7dd73913c.png)

k=14 plot.  N=1989 remaining points
![image](https://user-images.githubusercontent.com/23244088/217383650-fb6e12e0-1153-422f-b7f6-544329a72e12.png)